lib/Telemetry/src/Dispatcher.h

```diff
@@ -0,0 +1,60 @@
+#include "Particle.h"
+#include "DataQueue.h"
+#include "IntervalLogger.h"
+
+#ifndef _DISPATCHER_H_
+#define _DISPATCHER_H_
+
+/**
+ * Dispatcher owns and operates on a collection of loggers
+ **/
+class Dispatcher {
+    public:
+        /**
+         * Constructs a Dispatcher with loggers, numLoggers and DataQueue
+         **/
+        Dispatcher(IntervalLogger **loggers, uint16_t numLoggers, DataQueue *dataQ) {
+            _loggers = loggers;
+            _numLoggers = numLoggers;
+            _dataQ = dataQ;
+            _logThisLoop = false;
+        }
+
+        /**
+         *  Must be called from main loop!  Takes times since program start (in seconds) and checks whether
+         *  log needs to be called on any of its loggers.
+         * 
+         * @param time the time in seconds since the start of the program
+         **/
+        void run(unsigned long time) {
```

Comment:
Can we get the millis() value in this function instead of passing it in as a parameter?

Reply:
When I was using millis() instead of seconds, the loggers weren't actually overlapping properly so we were getting more time stamps in the published json than we needed (multiple time stamps with the same unix time).  Working with seconds fixed that, and now overlapping logs will always be published under the same time stamp.

I think it's because the loop doesn't overlap perfectly with millis(), so for example a 5s logger might get called to log at say 5001 instead of 5000 and their last log times would be set to this.  This means when the clock reaches 10 000 and all the 10s loggers are called, the 5s logger won't because it's still waiting for millis() 10 001

Reply:
i just realized your question wasn't about changing to millis().  I think particle.h is already included so yeah that could be fetched internally.  I'll change that

Reply:
Yep just getting it inside the library is what I meant. I understood what you meant about the second accuracy, and I think that's a good way of doing it right now that our target accuracy is 1s.

Reply:
this ^ and the data queue being passed into the constructor reminded me about dependency injection.  If we were writing unit tests for all his, it would be preferable to pass in any of the dispatcher's dependencies as arguments so that we could control the testing context https://stackoverflow.com/questions/130794/what-is-dependency-injection

Reply:
not necessarily relevant right now, but good to keep in mind